docs/creating-custom-templates.md

```diff
@@ -61,28 +57,57 @@ For example, you'd like to keep `version` the same, so in your dictionary you wi
 ```
 'version': 'version'
 ```
+## 2. Write a Custom Document Generator
 
-## 2. Write a Custom Document Generator (Optional)
-You could use the available yaml or json format, in which case you don't have to do anything more, or if you need some custom arrangement (for example, you want to arrange the keys and values in
-a different way. Then you can do that within the folder `tern/report`. For example, `spdxtagvalue` is one of the folders in here. Tern will automatically look for a module called `generator` in here.
+Tern uses a python module called Stevedore to dynamically load each report plugin at runtime. When Tern is invoked from the command line using the `report -f <format>` option,  it will use Stevedore to call the `generate()` function inside the specified report format. You can see how `generate()` gets called in the `generate_format` function inside `tern/report/report.py`.
 
-To make a generator, create a folder in `tern/report` containing the files:
+Because Tern loads the report plugin at runtime using Stevedore, it is required that you derive your own `generate()` function. This should happen inside your own subclass dervied from the `Generate` abstract base class in `tern/formats/generator.py`. To make a generator, create a folder under `tern/formats/<your_custom_report_name>`. For the purpose of this example, let's call the plugin you want to create `custom`.  You would create a new directory `tern/formats/custom` containing the files:
 ```
 __init__.py
 generator.py
 ```
 
-The `generator.py` should have a function called `generate.py` which takes in a list of Image objects. If you think you will be working with only one Image object, reference just the first image. From here you can use `to_dict(template)` which will take in your template object and return a dictionary containing all the keys definined in your mapping. For example:
+The `generator.py` file should have a function called `generate()` which takes two inputs: `self`, and a list of Image objects. If you think you will be working with only one Image object, reference just the first image. From here you can use `to_dict(template)` which will take in your template object and return a dictionary containn
```

Comment:
"working with only one Image object" -> This one seems to come out of nowhere. Maybe it needs to be removed.